clients/src/main/java/org/apache/kafka/common/serialization/StringDeserializer.java

```diff
@@ -27,27 +30,45 @@
  *  value.deserializer.encoding or deserializer.encoding. The first two take precedence over the last.
  */
 public class StringDeserializer implements Deserializer<String> {
+
```

Comment:
the extra line :)

Reply:
I see that some classes have blank line under the class, and some have no blank line, so what is the recommended style of kafka:
* have blank line: ByteArrayDeserializer, DoubleDeserializer, ListDeserializer;
* have no blank line: ByteArraySerializer, ByteBufferSerializer, LongDeserializer

Reply:
Usually I like to follow [Google's java style guide](https://google.github.io/styleguide/javaguide.html#s4.5-line-wrapping).  In this case, no new line after an opening brace.  Perhaps we can get one of the committer to comment on the styling.

Reply:
We don't really have a rule for this. It's usually best practice to avoid changing a lot of spacing in a PR (if it's not required) to reduce noise.

Reply:
> We don't really have a rule for this. It's usually best practice to avoid changing a lot of spacing in a PR (if it's not required) to reduce noise.

Ok, I got it.